app/controllers/dashboard/work_versions_controller.rb

```diff
@@ -37,6 +37,12 @@ def show
     def edit
       @work_version = policy_scope(WorkVersion).find(params[:id])
       authorize(@work_version)
+
+      my_creator = Creator.find_or_create_by_user(current_user)
+      my_creator.readonly!
+      if @work_version.creator_aliases.none? { |ca| ca.creator == my_creator }
+        @work_version.creator_aliases.build(alias: my_creator.name, creator: my_creator)
+      end
```

Comment:
100% yes, and I was thinking the same thing as you, I just didn't want to go totally hardcore with the factoring if the underlying data model turned out to be bogus. But I'm glad you said this because I was thinking the same thing